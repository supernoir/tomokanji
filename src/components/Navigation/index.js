import React from 'react';
import Button from '../Button';

const Navigation = props => {
	return (
		<aside className="nav">
			<Button label={'‹ 前 Previous'} handleClick={props.handlePrevious} className={'button'} />
			<Button label={'Random 運'} handleClick={props.handleRandom} className={'button random'} />
			<Button label={'Next 次 ›'} handleClick={props.handleNext} className={'button'} />
		</aside>
	);
};

export default Navigation;
