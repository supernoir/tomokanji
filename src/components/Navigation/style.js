import styled from 'styled-components';
import { Navigation } from './index';

export const StyledNavigation = styled(Navigation)`
  margin: 2em 0 4em;
  display: flex;

  @media screen and (min-width: 480px) {
    flex-direction: row;
  }

  @media screen and (max-width: 480px){
    flex-direction: column;
  }
`;
