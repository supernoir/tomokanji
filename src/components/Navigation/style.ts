import styled from 'styled-components';

export const StyledNavigation = styled.aside`
  margin: 2em 0 4em;
  display: flex;
  width: 100%;
  justify-content: center;
  
  @media screen and (min-width: 480px) {
    flex-direction: row;
  }

  @media screen and (max-width: 480px){
    flex-direction: column;
  }
`;
