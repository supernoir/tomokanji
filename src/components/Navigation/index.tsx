import React from 'react';
import { StyledButton, StyledRandomButton } from '../Button/style';
import { StyledNavigation } from './style';

interface NavigationProps {
	handlePrevious: () => void;
	handleRandom: () => void;
	handleNext: () => void;
	children?: JSX.Element;
}

const Navigation: React.FC<NavigationProps> = (props) => {
	return (
		<StyledNavigation>
			<StyledButton label={'‹ 前 Previous'} handleClick={props.handlePrevious} />
			<StyledRandomButton label={'Random 運'} handleClick={props.handleRandom} />
			<StyledButton label={'Next 次 ›'} handleClick={props.handleNext} />
		</StyledNavigation>
	);
};

export default Navigation