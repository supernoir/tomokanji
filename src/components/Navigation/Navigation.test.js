import React from 'react';
import Navigation from './index';
import { render, shallow } from 'enzyme';

describe('Navigation', () => {
	const testClickHandler = jest.fn();
	test('should match the snapshot', () => {
		const component = render(<Navigation handlePrevious={testClickHandler} handleNext={testClickHandler} handleRandom={testClickHandler} />);
		expect(component).toMatchSnapshot();
	});
});
