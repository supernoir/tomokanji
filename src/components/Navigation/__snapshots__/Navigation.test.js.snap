// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
<aside
  class="nav"
>
  <a
    class="button"
  >
    ‹ 前 Previous
  </a>
  <a
    class="button random"
  >
    Random 運
  </a>
  <a
    class="button"
  >
    Next 次 ›
  </a>
</aside>
`;
